src/Symfony/Component/Process/Process.php

```diff
@@ -291,20 +291,22 @@ public function start(callable $callback = null, array $env = [])
         $this->hasCallback = null !== $callback;
         $descriptors = $this->getDescriptors();
 
+        if ($this->env) {
+            $env += $this->env;
+        }
+
         if (\is_array($commandline = $this->commandline)) {
             $commandline = implode(' ', array_map([$this, 'escapeArgument'], $commandline));
 
             if ('\\' !== \DIRECTORY_SEPARATOR) {
                 // exec is mandatory to deal with sending a signal to the process
                 $commandline = 'exec '.$commandline;
             }
+        } else {
```

Comment:
I would call this only on Windows since the method does nothing on Linux

Reply:
I've change to preg_match to allow throwing even on linux if the place are present but the value are not passed, this was asked in the previous PR.

Reply:
back to the callback, but I think we should still check on linux since it adds a little DX when the envvar does not exist

Reply:
I don't understand what's your concern here. This already happens on Linux, right?

Reply:
It does not throw, it shows nothing if the envvar are missing

Reply:
Why do tests pass then?

Reply:
What I meant is, If I don't add the throw for linux, it does not work and shows nothing instead of the placeholders.
With this, it says that no value exists for the given placeholder, maybe this behaviour should not exist ?

Reply:
Ok all good, I'm fine with the exception. Thanks.